### DISCUSSION TITLE
Add user role as a CSS class to HTML container

### DISCUSSION THREAD

**Discussion by aaron-manning** [directus/directus#10942] [Feature Requests]
Can we get the user's role output as a class in the HTML?

It would make it possible to visually hide features for certain roles, which seems to be impossible after the modules config was changed to global rather than per-role.

Examples:
```
<div id="directus" class="directus-role-roleName">
<div id="directus" class="directus-role-administrator">
<div id="directus" class="directus-role-editor">
```

**Comment by benhaynes** [comment]
We've discussed this quite a bit internally, and if this was done it would need to be completely unofficial. Removing or changing these internal app CSS selectors would never be considered a breaking change.

**Reply by aaron-manning** [reply]
By completely unofficial, do you mean the team needs a PR?

I'm happy to submit a PR if this will get merged in. I'm sure there must be more uses for this feature than my own :)

**Reply by benhaynes** [reply]
Haha, well, PRs are always appreciated... and will certainly speed up a merge! But no, "unofficial" just means that it will be flagged as such in the docs so that users know that those selectors may change at any time and are not considered a breaking change.

Still, unofficial or not, I'd like to discuss the plans for these IDs and classes here before implementing.  That will at least give us more time before they might be changed again.

CC @rijkvanzanten

**Reply by rijkvanzanten** [reply]
I'm not sure about using the role name as the class. If anything, we should rely on the UUID as that doesn't change 🤔 

In any case, what @benhaynes mentioned is correct. We wouldn't consider this an official API for theming, and it might change at any time.

**Reply by benhaynes** [reply]
Exactly, we need to use the immutable keys for these... not names. And therefore should prefix things so it's clear what it is. Eg:

```
.current-user-[uuid]
.current-role-[uuid]
.collection-[collection-key]
.field-[field-key]
```

Maybe we even want to add a `-selector` suffix to these to better differentiate between what's officially used, and these unofficial selectors?

**Reply by rijkvanzanten** [reply]
I'd rather think through more official theming strategies, where this could be a part of it, than add in arbitrary classes left and right that aren't officially supported for this purpose 😬

**Reply by benhaynes** [reply]
I agree! Formally supporting all the things that people want to do is the ideal, and that should still be the priority. But until that is done, having unofficial selectors that allow some manual overrides is a decent stop-gap. Also, adding a few _key_ selectors like the ones above is nice either way since we have the Custom CSS in settings. Again, I am only sanctioning the four I mentioned so that things don't get out of hand.

User and Role would just be on the body or app element, and the other two on their respective collection/field wrapper elements.

**Reply by Pachat** [reply]
Another way to obtain a role customization is to add a `Custom CSS` field to the roles.

That `Custom CSS` would be loaded **after** the `Project Settings Custom CSS` allowing to supercharge it's basic CSS rules.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
